src/serac/physics/heat_transfer.hpp

```diff
@@ -328,17 +380,36 @@ class HeatTransfer<order, dim, Parameters<parameter_space...>, std::integer_sequ
   /**
    * @brief Set the thermal flux boundary condition
    *
-   * @tparam FluxType The type of the flux function
-   * @param flux_function A function describing the thermal flux applied to a boundary
+   * @tparam FluxType The type of the thermal flux object
+   * @param flux_function A function describing the flux applied to a boundary
+   *
+   * @pre FluxType must be a object that can be called with the following arguments:
+   *    1. `tensor<T,dim> x` the spatial coordinates for the quadrature point
+   *    2. `tensor<T,dim> n` the outward-facing unit normal for the quadrature point
+   *    3. `double t` the time (note: time will be handled differently in the future)
+   *    4. `T temperature` the current temperature at the quadrature point
+   *    4. `tuple{value, derivative}`, a variadic list of tuples (each with a values and derivative),
+   *            one tuple for each of the trial spaces specified in the `DependsOn<...>` argument.
+   *
+   * @note The actual types of these arguments passed will be `double`, `tensor<double, ... >` or tuples thereof
+   *    when doing direct evaluation. When differentiating with respect to one of the inputs, its stored
+   *    values will change to `dual` numbers rather than `double`. (e.g. `tensor<double,3>` becomes `tensor<dual<...>,
+   * 3>`)
```

Comment:
Just noticing that, because of the nice re-usable form of `Functional`, this comment is boilerplate that we must maintain separately for each physics module. Is there a way we could write this comment once (suitably genericized) and have it appear over these calls? I'm not a doxygen expert.

Reply:
Not sure. I'll look into it. However, these call lists will be subtlety different for each physics constitutive response/load, so it might be worth the repetition.

Reply:
It looks like you can, but you have to use the entire docstrings. See [here](https://doxygen.nl/manual/commands.html#cmdcopydoc).

Reply:
That's a good find. We could replace "temperature" with "primal field", etc. in that block and only lose a tiny bit of specificity. Let's shelve that for now, but keep it in mind.